python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy
+
+import lsstDebug
+import lsst.afw.display.ds9 as ds9
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+__all__ = ("SubtractBackgroundConfig", "SubtractBackgroundTask")
+
+class SubtractBackgroundConfig(pexConfig.Config):
+    """!Config for SubtractBackgroundTask and lsst::afw::math::makeBackground
+
+    @warning This config is automatically converted to an lsst::afw::math::BackgroundControl
+    for lsst::afw::math::makeBackground, so do not change the field names or types.
```

Comment:
This seems fragile, especially since `BackgroundControl` is in a separate package.  How about moving this `SubtractBackgroundConfig` into afw as `BackgroundConfig`?  If necessary, you could subclass it here.  That would keep together things that go together.